Tests/IntegrationTests/PSQLIntegrationTests.swift

```diff
@@ -360,4 +360,22 @@ final class IntegrationTests: XCTestCase {
         }
     }
     
+    func testConnectionClosureMidQueryDoesNotHang() async throws {
+        _ = await withThrowingTaskGroup(of: Void.self) { taskGroup in
+            for _ in (0 ..< 1_000) {
+                taskGroup.addTask {
+                    let conn = try await PostgresConnection.test(
+                        on: NIOSingletons.posixEventLoopGroup.next()
+                    ).get()
+
+                    async let close: () = conn.closeGracefully()
+                    async let query = conn.query("SELECT 1", logger: .psqlTest)
```

Comment:
In which case do we run into a problem? Reading the code in `PostgresConnection` I assume we always run into the issue if we try to execute a query after the connection was closed?

Reply:
Not quite after. If the channel is already closed you get the "on IO Closed" error.

Reply:
I was wrong as mentioned below.